chatterbot/utils/parsing.py

```diff
@@ -0,0 +1,491 @@
+import re
+import calendar
+from datetime import timedelta, date, datetime
+
+"""
+The first step is the tag the text with <TIMEX>
+1. Extract => The meeting is 25 minutes from now => The meeting is <TIMEX>25 minutes from now<TIMEX>
+2. Convert to datetime
+"""
+
+"""
+Regular expression matches
+https://github.com/nltk/nltk_contrib/blob/master/nltk_contrib/timex.py
```

Comment:
Apache 2.0 requires you to, at the very minimum (probably more to it), include the license text in the files which are licensed under Apache 2.0.

You would be better off optionally depending on NLTK for any adapters which need date parsing support.